Sources_v3/Database/DTOs/CurrentUserDTO.swift

```diff
@@ -83,6 +84,42 @@ extension NSManagedObjectContext: CurrentUserDatabaseSession {
         dto.unreadMessagesCount = Int16(count.messages)
     }
     
+    func saveCurrentUserDevice(_ device: DevicePayload) throws {
+        guard let currentUser = currentUser() else {
+            throw ClientError.CurrentUserDoesNotExist()
+        }
+        
+        saveDevice(device, for: currentUser)
+    }
+    
+    func saveCurrentUserDevices(payload: DevicesPayload) throws {
+        try saveCurrentUserDevices(payload.devices)
+    }
+    
+    private func saveCurrentUserDevices(_ devices: [DevicePayload]) throws {
+        guard let currentUser = currentUser() else {
+            throw ClientError.CurrentUserDoesNotExist()
+        }
+        
+        currentUser.devices.removeAll()
+        
+        for device in devices {
+            saveDevice(device, for: currentUser)
+        }
+    }
+    
+    private func saveDevice(_ device: DevicePayload, for currentUser: CurrentUserDTO) {
+        let dto = DeviceDTO.loadOrCreate(id: device.id, context: self)
+        dto.createdAt = device.createdAt
+        dto.user = currentUser
+    }
```

Comment:
I don't think we need all these overloads. One simple method `saveCurrentUserDevices(_ devices: [DevicePayload])` should be enough.

Reply:
Converted it to one function in 107624943165ad39afd991a7fd9aba8d58c1a28a